client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -834,19 +834,24 @@ public int pushOrMergeData(
     // increment batchId
     final int nextBatchId = pushState.nextBatchId();
 
-    // compress data
-    final Compressor compressor = compressorThreadLocal.get();
-    compressor.compress(data, offset, length);
+    int totalSize = data.length;
+    byte[] shuffleDataBuf = data;
 
-    final int compressedTotalSize = compressor.getCompressedTotalSize();
+    if (conf.shuffleCompressionEnabled()) {
```

Comment:
We'd better add a variable for this to avoid frequent call.

Reply:
@AngersZhuuuu done.